fuzz/Makefile

```diff
@@ -9,12 +9,17 @@ FUZZ_JOBS?=$(shell nproc)
 check:
 	reuse lint
 
+corpus:
+	7za x $<
```

Comment:
I think we should have the corpus be a dedicated folder. Why zip it? This prevents it from efficiently showing up in the diffs.

There should probably be a way to regenerate the corpus with a new format if we ever change how the input is parsed. Opcard does it for an arbitrary `Input` structure that also contains config flags.

Reply:
1. My corpus right now is 17MB (out of the initial 300kB of raw binary requests), and quite too big for a repo, if assumed to be changing during the course of development. Aside the size, I am not sure there is much sense for versioning it. Unless, do you propose to not include there the corpora found during the fuzzing?
2. The corpus was generated from the real requests, gathered during the pynitrokey's complex tests execution (single corpus per test case; see the linked [PR](https://github.com/Nitrokey/pynitrokey/pull/296)). Certainly the description of generating it is needed. Will be added in the next commits.
3. Arbitrary makes sense in more complex cases. For now I do not have any needs for flags, but maybe adding a switchable PIN check would make sense? I still have the required coverage with the current corpus though. I would essentially need to reformat the Python taken samples to Arbitrary format, while preparing for fuzzing (it seems though I have just made it working faster, avoiding the internal loop - to be uploaded). The alternative is to rewrite the test cases from Python to Rust, or make some less complicated scenarios only. For the latter I still believe this is a redundant work when we have the test cases written already.

Reply:
Also to do:
- [ ] fix corpus target (add the missed dep in the recipe)

Reply:
Given the discussion we had I think it's fine if we keep it that way, but I'd rather:

- Use tar.gz as the archive format since the cli tool are more widely installed
- Document the stable function: mention the link to the seed corpus and that if one is changed the other should be too to prevent them from getting out of sync.
- Document how to regenerate the corpus.

Reply:
Corrected as requested. 
The doc file is now here: https://github.com/Nitrokey/oath-authenticator/tree/main/fuzz#fuzzing. In case you see its missing anything please let me know.